app/src/main/java/com/otaman/weather/networking/CurrentWeatherNetworkMapper.kt

```diff
@@ -11,4 +25,30 @@ fun CurrentWeatherResponse.toCurrentWeather(): CurrentWeather {
         conditionIcon = "https:${current.condition.iconCondition}",
         conditionText = current.condition.textCondition
     )
+}
+
+@RequiresApi(Build.VERSION_CODES.O)
+fun ForecastResponse.toForecast(): List<ForecastWeather> {
+    return forecast.forecastDay.map { item ->
+        val localDate = LocalDate.parse(item.date)
+        val month = localDate.month.toString()
+        val day = localDate.dayOfMonth.toString()
+
+        ForecastWeather(
+            dayTemp = item.day.dayTemp,
+            date = "$day $month",
+            dayConditionIcon = "https:${item.day.dayCondition.dayConditionIcon}",
+            hourly = item.hour.map { hour ->
+                hour.toHourly()
+            }
+        )
+    }
+}
+
+fun Hour.toHourly(): HourlyWeather {
+    return HourlyWeather(
+        hourlyTemp = temp,
+        hourlyConditionIcon = "https:${condition.conditionIcon}",
+        time = time.drop(11)
```

Comment:
Is this right way to format time like this?
Time pattern which we receive from API is `"yyyy-MM-dd HH:mm"`, I tried to format it to `"HH:mm"`, but unsuccessfully. So I decided to just drop first 11 symbols from received string (e.g. `"2022-09-13 17:00"`) and as a result we have string `"17:00"`

Reply:
This is definitely not a right way. You must use date pattern